ocs_ci/ocs/bucket_utils.py

```diff
@@ -2739,3 +2794,89 @@ def map_objects_to_owners(mcg_obj, bucket_name, prefix=""):
     """
     response = s3_list_objects_v2(mcg_obj, bucket_name, prefix=prefix, fetch_owner=True)
     return {item["Key"]: item["Owner"] for item in response.get("Contents", [])}
+
+
+def sts_assume_role(
+    pod_obj,
+    role_name,
+    access_key_id_assumed_user,
+    role_session_name=None,
+    mcg_obj=None,
+    signed_request_creds=None,
+):
+    """
+    Aws s3 assume role of an User
+
+    Args:
+        role_name (str): Role name of a role attached to the assumed user
+        access_key_id_assumed_user (str): Access key id of the assumed user
+        mcg_obj (MCG): MCG object
+        signed_request_creds (dict): a dictionary containing AWS S3 creds for a signed request
+    Returns:
+        Dict: Representing the output of the command which on successful execution
+        consists of new credentials
+
+    """
+    if not role_session_name:
+        role_session_name = f"role-session-{uuid4().hex}"
+    cmd = (
+        f"assume-role --role-arn arn:aws:sts::{access_key_id_assumed_user}:role/{role_name} "
+        f"--role-session-name {role_session_name}"
+    )
+    cmd = craft_sts_command(
+        cmd, mcg_obj=mcg_obj, signed_request_creds=signed_request_creds
+    )
+    return pod_obj.exec_cmd_on_pod(command=cmd)
+
+
+def s3_create_bucket(s3_obj, bucket_name, s3_client=None):
+    """
+    AWS s3 create bucket
+
+    Args:
+        s3_obj (MCG): MCG object
+        bucket_name (str): Name of the bucket
+        s3_client (S3.Client): Any S3 client resource
+
+    """
+    if s3_client:
+        return s3_client.create_bucket(Bucket=bucket_name)
```

Comment:
according to this code, s3_obj is a mandatory arg and the bucket can be created with it. If that is case, why do you need s3_client?

Reply:
s3_obj consists of admin credentials. but if any function call passes s3_client (optional) parameter then we use that s3_client instead of s3_obj. s3_client will have credentials for any customer s3 user